### DISCUSSION TITLE
[BP#3] Fluidifier le processus

### DISCUSSION THREAD

**Discussion by ddebarbieux** [cnumr/best-practices#114] [Bonnes pratiques] [answered]
Ajouter aux exemples l'achat en un clic d'Amazon qui est un modèle de processus fluide centré sur l'essentiel.

En tant qu'utlisateur, j'aimerai voir plus souvent ce concept 'action complexe réalisée en un clic'.

(si quelqu'un a un autre exemple qu'Amazon, je prends aussi)

**Comment by ACTLEM** [comment]
Je suis pour ajouter d'autres exemples, mais j'aimerais un autre qu'Amazon :-D

**Reply by brpasi** [reply]
Uber fait de même il me semble, mais il est vrai que sur cet exemple précis c'est Amazon le précurseur.
A noter que cela ne va pas sans quelques effets collatéraux / risques : sécurité, augmentation du nb d'achat "impulsif" (> Surconso d'équipement)...

**Comment by fzaninotto** [comment]
Je ne comprends pas cette pratique, ni comment valider qu'elle est ou pas prise en compte.

**Reply by Trankilloman** [reply]
Cette pratique revient à dire qu'il faut s'assurer que : le service répond de manière très efficace à l'unité fonctionnelle (acheter LE produit sélectionné et le service ne répond pas à x ou y fonctions secondaires (proposer d'autres produits) voir parfois vraiment peu nécessaires POUR L'UTILISATEUR final (proposer des publicités)

**Reply by fzaninotto** [reply]
comment on mesure "très efficace" ? Comment on mesure qu'un service qui réponde de manière "momins efficace" a un plus gros impact ?

**Reply by florinesueur** [reply]
Il me semble qu'ici, il est question de l'efficacité du parcours utilisateur. Dans BP, la validation se fait avec une comparaison avec un site concurrent par rapport aux nombres d'étapes à réaliser. 

A une époque, l'exemple de l'achat d'un ticket de train sur SNCF et sur Captain Train (qui est devenu Trainline). De mémoire pour captain train, il fallait 4 screens pour acheter son billet et pour SNCF au moins 7 screens.
Malheureusement, nous ne pouvons pas prendre l'exemple de Sci-hub et de tous les sites universitaires de recherches d'articles scientifiques.

Il est difficile de trouver un exemple, car demain le parcours sera différent. L'UX étant de plus en plus prit en considération.

Du coup , si j'ai bien compris, ne serait-il pas intéressant de renommer cette BP en : Fluidifier le parcours utilisateur ?

**Reply by ACTLEM** [reply]
Pour compléter, dans quasiment toutes les ACV de service numérique basés sur une expérience web, le deuxième post en terme d'impacts environnementaux (le premier est la fabrication des terminaux) est le temps passé par l'utilisateur sur le terminal. Réduire le temps de parcours est donc un levier important.
Perso, je ne suis pas contre remplacer "processus" par "parcours utilisateur".

**Reply by ACTLEM** [reply]
A voir si la navigation/parcours utilisateur peut être repris par la proposition https://github.com/cnumr/best-practices/discussions/89.
Et plus insister sur la bonne pratique existante sur la partie formulaire.
On aurait deux bonnes pratiques: celle-ci axée sur le contenu d'un formulaire et une autre sur le parcours utilisateur.

**Comment by Christophe-Cl** [comment]
Suggestion "Optimiser le parcours utilisateur"
En se basant sur les analytics et les méthodes d'observation UX (recherches, tests, questionnaires, sondages, etc.) le but est de réduire le temps d'utilisation de l'unité fonctionnelle d'un service et le parcours des utilisateurs cherchant à accomplir une tache. La BP est donc inverse aux méthodes des réseaux sociaux par exemple, ou de tout service numérique usant des dark-patterns afin de capter l'attention.

**Reply by fzaninotto** [reply]
Pourquoi pas un titre encore plus explicite: "Raccourcir la durée des parcours utilisateurs", parce que pour optimiser, il faut savoir quelle métrique regarder...

**Comment by ACTLEM** [comment]
Côté équipe projet, nous sommes pour modifier en insistant plus sur la réduction de la durée du parcours utilisateur, voir mettre des exemples concrets. Etes-vous en phase?

**Reply by Christophe-Cl** [reply]
+1. Toutefois j'ai encore un léger doute sur le terme "durée" qui est lié au temps passé par l'utilisateur. L'objectif est surtout de réduire les requêtes et les calculs inutiles au sein d'un parcours optimisé pour l'utilisateur autour de son besoin et de l'unité fonctionnelle qui y répond efficacement (pour ne pas écrire "qui y répond au plus vite" en lien avec la "durée" :) ...)

**Reply by ACTLEM** [reply]
Désolé, je n'ai pas compris ta remarque ^^
Le but final est de réduire le temps passé par l'utilisateur pour utiliser le service, je ne comprends pas les mentions de requêtes et de calculs pour cette bonne pratique fonctionnelle.

**Reply by florinesueur** [reply]
En fait, si j'ai bien compris; il y a deux visions : 
- la durée du parcours uniquement
- le nombre de wireframes (ou screens ou pages ou étapes ou actions) pour faire le parcours

Il me semble qu'on peut dire que le bouton amazon (que je ne connais pas) permet d'économiser du temps à l'utilisateur, mais aussi x étapes du parcours avec formulaire et surement publicité qui permet d'économiser x requêtes, x images et x nouvelles mises en page.

**Reply by ACTLEM** [reply]
C'est ce que dit déjà la BP, non? Le principal problème sur cette BP reste l'exemple (ou l'absence d'autre exemple) finalement de mon point de vue.

**Reply by Christophe-Cl** [reply]
(si je comprends bien) Le but n'est pas tant de réduire le temps passé par l'utilisateur, donc la durée de son utilisation, mais de réduire le nombre de requêtes, de pages, d'écrans, donc de ressources consommées. 
Exemple : Je mets 5 minutes à lire 1 page article (VS) je passe 5 minutes et génère 50 pages à trouver mon article => la durée importe peu, en revanche le parcours, l'ergonomie, l'UX et l'optimisation autour de l'unité fonctionnelle pour la trouver facilement comptent beaucoup plus.

**Reply by ACTLEM** [reply]
La BP dit les deux: limiter le nombre d'étapes et de pages tout en évitant qu'elles soient trop complexes.

**Reply by Usocrate** [reply]
Salut à tous, je suis ergonome depuis 20 ans. Demander de raccourcir le nombre d'étapes ou la durée du parcours utilisateur dans l'absolu me semble difficilement applicable. En ergonomie nous avons un concept très utile : **le dévoilement progressif**. 
Ça consiste à n'exposer  à l'écran que les fonctionnalités / données essentielles dans une premier temps pour ne donner accès aux autres que dans un second temps, à la demande. Un exemple : lorsque vous ajoutez un événement dans Google Calendar, la boîte de dialogue propose les champs essentiels (titre, date, lieu, invités,...) et les fonctionnalités plus avancées sont accessibles sur un autre écran après clic sur un bouton "Autres options" (paramétrage de notifications, privilèges des invités, caractère privé de l'événement,...).
Si le dévoilement progressif devenait une bonne pratique, la validation pourrait reposer sur le taux d'utilisation des fonctionnalités exposées.
[nngroup.com/articles/progressive-disclosure](https://www.nngroup.com/articles/progressive-disclosure/)

**Reply by ACTLEM** [reply]
Merci beaucoup @Usocrate. Je réagis à chaud: Le dévoilement progressif pourrait effectivement devenir une bonne pratique ou être intégré à cette BP (https://github.com/cnumr/best-practices/blob/main/chapters/BP_003_fr.md). Réduire le temps d'utilisation du service est un but mais est sûrement trop générique, d'où les discussions. Avoir des BP plus concrètes, terrain pour compléter/remplacer serait sûrement plus opérationnel.

**Reply by florinesueur** [reply]
J'ai retrouvé une vieille présentation avec les parcours pour acheter un billet de train : https://docs.google.com/presentation/d/1sgpXwxaY58h89arjq8UeXMUyCnZB2dCyEJjlXWX5y3s/edit?usp=sharing (il faut faire play)

Aussi d'autres exemples qui me passent en tête :
- "achat express" qui dispense d'aller sur la page produit comme la Fnac
- les infobulles de wikipedia donnent un aperçu de la page avec des fois assez d'informations pour éviter d'aller sur la page
- lien transversal : le fait de pourvoir relier une issue à une pull request, évite d'afficher tout le listing des pull requests
- mettre dans les favoris au lieu de refaire la recherche ou encore les listes (amazon, ...) d'achat ou de courses (en un click je recommande ma liste de course habituelle)
- l'utilisation de template préétabli (Miro, notion , Clickup, ...) ou faire ses templates
- préremplir les formulaires, par exemple, le contrôle technique avec la plaque d'immatriculation ou de préremplir avec les méta d'un article (https://news.humancoders.com/) ou préremplir les infos d'une entreprise avec l'api Insee permettant aussi d'éviter les erreurs de la part de l'utilisateur
- acheter sans inscription dans un e-commerce
- les raccourcis clavier peuvent éviter d'ouvrir un menu

Mais en effet, ce n'est pas aux dépens du développement progressif qui est une bonne BP.
Aussi le parcours dépend du business model (vive la pub).

Au lieu de fluidifier peut-être que c'est simplifier (ou faciliter) le parcours utilisateur (ou la tâche de l'utilisateur ou de l'unité fonctionnelle ou la charge de travail de Bastien et Scapin) ?

**Reply by Christophe-Cl** [reply]
Merci Florine ! certains exemples que tu cites sont très intéressants pour l'optimisation du/des parcours.

**Reply by brpasi** [reply]
Bonjour,
En phase pour renommer avec quelque chose "simplifier le parcours utilisateur" comme proposé par Florine.
Parmi les exemples cité : 
- ok pour l'achat express, achat sans inscription, lien transversal, mise en favori
- le cas des infobulles et du préremplissage de formulaire est pertinent aussi mais à manier avec précaution : possibilité ici de faire pire que mieux dans certains cas (dans le cas de l'infobulle, tu peux être amené à afficher de l'info là ou il n'y aurait peut être pas eu de clic sur la page cible). 
- pour les raccourcis clavier, on sort peut être un peu du champ direct de ce type de BP

**Comment by ACTLEM** [comment, marked as answer]
Issue créée pour modification exemples et titre: https://github.com/cnumr/best-practices/issues/248
